src/plugins/intel_gpu/src/kernel_selector/kernels/rope/rope_kernel_base.cpp

```diff
@@ -22,28 +22,53 @@ JitConstants RoPEKernelBase::GetJitConstants(const rope_params& params, RoPEKern
         jit.AddConstant(MakeJitConstant("ENABLE_IO_COPY", true));
     }
 
+    if (params.gather_rank > 0) {
+        jit.AddConstant(MakeJitConstant("ENABLE_GATHER", true));
+        jit.AddConstant(MakeJitConstant("GATHER_RANK", params.gather_rank));
+    }
+
     if (params.slice_stop - params.slice_start > 0) {
```

Comment:
BTW, it looks like this check doesn't make much sense because both numbers are of size_t type, and the result would be true in all cases except when slice_stop is equal to slice_start